File: 대표 알고리즘/정렬/selection-sort.js
let array = [9, 8, 7, 6, 5, 4, 3, 2, 1]

const selectionSort = (array) => {
  for (let i = 0; i < array.length; i++) {
    let minIdx = i
    let tempVal = array[i]
    for (let j = i + 1; j < array.length; j++) {
      if (array[i] > array[j]) {
        minIdx = j
      }
    }
    array[i] = array[minIdx]
    array[minIdx] = tempVal
    console.log(array)
  }
}

console.log(`before : ${array}`)
selectionSort(array)
console.log(`after : ${array}`)

export {}

File: 대표 알고리즘/정렬/selection-sort.py
array = [9,8,7,6,5,4,3,2,1]

def selectionSort(array):
  n = len(array)
  for i in range(0, n):
    minIdx=i
    for j in range(i, n):
      if(array[minIdx]>array[j]):
        minIdx=j
    array[i], array[minIdx] = array[minIdx], array[i]
    print(array)


print("before: ", array)
selectionSort(array)
print("after: ", array)